tests/integration/test_dynamic_sampling.py

```diff
@@ -348,6 +348,69 @@ def test_it_tags_error(mini_sentry, relay, expected_sampled, sample_rate):
     assert evt_id == event_id
 
 
+def test_sample_on_parametrized_root_transaction(mini_sentry, relay):
+    """
+    Tests that dynamic sampling on traces operate on
+    the parametrized version of the root transaction.
+    """
+    project_id = 42
+    relay = relay(mini_sentry, _outcomes_enabled_config())
+
+    # The pattern used to parametrize the transaction
+    pattern = "/auth/login/*/**"
+    # How the original transaction is in the DSC.
+    original_transaction = "/auth/login/test/"
+    # What the transaction is transformed into, which the dynamic sampling rules should respect.
+    parametrized_transaction = "/auth/login/*/"
+
+    config = mini_sentry.add_basic_project_config(project_id)
+    public_key = config["publicKeys"][0]["publicKey"]
+
+    config["config"]["txNameRules"] = [
+        {
+            "pattern": pattern,
+            "expiry": "3022-11-30T00:00:00.000000Z",
+            "redaction": {"method": "replace", "substitution": "*"},
+        }
+    ]
+    config["config"]["transactionMetrics"] = {"version": 1}
+
+    ds = config["config"].setdefault("sampling", {})
+    ds.setdefault("version", 2)
+
+    sampling_rule = {
+        "samplingValue": {"type": "sampleRate", "value": 0.0},
+        "type": "trace",
+        "condition": {
+            "op": "and",
+            "inner": [
+                {
+                    "op": "eq",
+                    "name": "trace.transaction",
+                    "value": parametrized_transaction,
+                    "options": {
+                        "ignoreCase": True,
+                    },
+                }
+            ],
+        },
+        "id": 1,
+    }
+
+    rules = ds.setdefault("rules", [sampling_rule])
+    rules.append(sampling_rule)
+
+    envelope = Envelope()
+    transaction_event, trace_id, _ = _create_transaction_item()
+    envelope.add_transaction(transaction_event)
+    _add_trace_info(envelope, trace_id, public_key, transaction=original_transaction)
```

Comment:
We should add a second project config that contains the txNameRules and pass its project key here, to test that the normalization uses the correct config.